erts/emulator/beam/jit/arm/instr_arith.cpp

```diff
@@ -1206,11 +1218,21 @@ void BeamModuleAssembler::emit_i_bsl(const ArgLabel &Fail,
                                      const ArgSource &LHS,
                                      const ArgSource &RHS,
                                      const ArgRegister &Dst) {
-    Label generic = a.newLabel(), next = a.newLabel();
     auto [lhs, rhs] = load_sources(LHS, ARG2, RHS, ARG3);
     auto dst = init_destination(Dst, ARG1);
 
+    if (is_bsl_small(LHS, RHS)) {
+        comment("skipped tests because operands and result are always small");
+        a.and_(TMP1, lhs.reg, imm(~_TAG_IMMED1_MASK));
+        a.lsl(TMP1, TMP1, imm(RHS.as<ArgSmall>().getSigned()));
```

Comment:
Isn't it possible for `RHS` to be a register here?

```suggestion
    if (RHS.isSmall()) {
        a.lsl(TMP1, TMP1, imm(RHS.as<ArgSmall>().getSigned()));
    } else {
        a.lsl(TMP1, TMP1, rhs.reg);
    }
```